client.go

```diff
@@ -148,6 +172,16 @@ func NewClient(options ClientOptions) (*Client, error) {
 	return &Client{namenode: namenode, options: options}, nil
 }
 
+func (c *Client) datanodeDialFunc(token *hadoop_common.TokenProto) func(ctx context.Context, network, addr string) (net.Conn, error) {
+	return (&rpc.DatanodeSaslDialer{
+		Dialer:    c.options.DatanodeDialFunc,
+		KeyFunc:   c.namenode.GetEncryptionKeys,
+		Privacy:   c.options.EncryptedDataNode,
+		Integrity: c.options.SecureDataNode,
+		Token:     token,
+	}).DialContext
+}
```

Comment:
Can we avoid wrapping it at all if those options are false?

Reply:
If you look at the definition for `DialContext` in the `DatanodeSaslDialer` it indeed will just call `Dial` and won't wrap the connection if those values are false. If you'd prefer though I can have this just return the DatanodeDialFunc if those values are false rather than using the sasl dialer